### PR TITLE
Improve docstrings and type hints (Part 3)

### PR DESCRIPTION
**Pull request by desilinguist**
- chore: ignore E501 errors in various files. 
- chore: fix docstrings and types in `skll.experiments` and `skll.learner` modules. 

To review this, please look at the [built documentation](https://skll.readthedocs.io/en/647-improve-docstrings-typehints-3/index.html) and check that under "API documentation",  look at the sections "`experiments` Package" and "`learner` Package" and confirm that:

- All parameters have a docstring.
- All type hints have clickable types or type hints.
- The default values are all specified as default = blah.
- All other changes look okay.

This is part of #647.

### CODE REVIEW

**Review by mulhod — APPROVED** [review]

File: skll/learner/__init__.py [hunk 1]
```diff
@@ -693,7 +682,8 @@ def get_feature_names_out(self) -> np.ndarray:
         Raises
         ------
         ValueError
-            If ``self.feat_vectorizer`` is either ``None`` or a ``FeatureHasher``.
+            If ``self.feat_vectorizer`` is either ``None`` or a
+            :class:`sklearn.feature_extraction.FeatureHasher``.
```
Comment (damien2012eng) [inline skll/learner/__init__.py:686]:
```suggestion
            :class:``sklearn.feature_extraction.FeatureHasher``.
```

Comment (desilinguist) [reply skll/learner/__init__.py:686]: No, when you specify roles, you only use single quotes.

Comment (damien2012eng) [reply skll/learner/__init__.py:686]: Okay. Should it be the single quote in the end as well? Here is what it shows right now. 
image

Comment (desilinguist) [reply skll/learner/__init__.py:686]: Oh I missed that! Sorry! I will fix it in the next PR!

**Pull request merged**